foundations/src/telemetry/memory_profiler.rs

```diff
@@ -56,12 +59,12 @@ struct Seal;
 /// A safe interface for [jemalloc]'s memory profiling functionality.
 ///
 /// [jemalloc]: https://github.com/jemalloc/jemalloc
-#[derive(Copy, Clone)]
+#[derive(Clone)]
 pub struct MemoryProfiler {
     _seal: Seal,
 
-    #[cfg(feature = "security")]
-    sandbox_profiling_syscalls: bool,
+    request_heap_profile: mpsc::Sender<oneshot::Sender<anyhow::Result<String>>>,
+    _heap_profiling_thread_handle: Arc<Mutex<JoinHandle<()>>>,
```

Comment:
I wasn't sure what to do with this JoinHandle, since we anticipate the thread living for the whole life of the application.

Should we just drop the handle and detach the thread? The thread will still die once all mpsc::Senders are dropped.

Reply:
I can only think of the use case of possibly observing a catastrophic failure of that thread (such as a panic) that causes it to end?

could it be useful to pass the handle to a spawned tokio task that joins the handle to observe the result of the thread?  that way if it fails unexpectedly we can log an error and it becomes observable, whereas in the thread itself we may not be able to handle unexpected panics?

Reply:
Side-note:  why do you have a Mutex around the handle?

Reply:
Having something else join the thread feels rather high-cost for a (hopefully) niche scenario, but it's probably better to have it in case the profiling thread does happen to panic. Unfortunately, it's a std `JoinHandle` which means joining it is a blocking operation. So this other watchdog task I think would actually have to be another std thread, not a tokio task. That, or we do some kind of polling on whether the heap profiling thread is finished. That doesn't seem great either. I'll see how it looks to have another watchdog thread.

The Mutex is superfluous in the current impl, and will be with the watchdog as well, I'll remove it

Reply:
I pursued a few avenues to join the profiling thread (using another thread, doing it when we notice sending to it fails), but they were all rather hairy. And, I believe, unnecessary. When a detached thread panics it still invokes the global panic hook, meaning it will write to stderr. In Oxy, we also extend the default panic hook by writing it to our logs with context, so a panic in the profiling thread will be logged. This seems sufficient to me.

Reply:
Good point, we just got to detach it.